### PR TITLE
fix: bump new release for JOSS

### PR DESCRIPTION
**Pull request by nicrie**
Following the [post review checks](https://github.com/openjournals/joss-reviews/issues/6060#issuecomment-1865186684) for the JOSS submission to
- update package author list and repository address. @slevang do you want to add an email address to the `pyproject.toml`?
- force new release for JOSS

### CODE REVIEW

**Review by slevang — APPROVED** [review]
I added an email, and also did a find/replace for the updated repo path.

FYI the new `nbstripout` hook wanted to remove all the notebook outputs, so I just committed with `--no-verify`. I assume you wanted to make sure these would all run properly in the docs build or something before stripping all of them?

File: pyproject.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "xeofs"
 version = "2.2.3"
 description = "Comprehensive EOF analysis in Python with xarray: A versatile, multidimensional, and scalable tool for advanced climate data analysis"
-authors = ["Niclas Rieger <niclasrieger@gmail.com>", "Samuel J. Levang"]
+authors = ["Niclas Rieger <niclasrieger@gmail.com>", "Sam Levang <slevang@gmail.com>"]
```
Comment (nicrie) [inline pyproject.toml]: I notice that your name here and in [paper.md](https://raw.githubusercontent.com/xarray-contrib/xeofs/joss-submission/publication/paper.md) is slightly different. Due to the JOSS review list:

> Make sure that the title and author list (including ORCIDs) in the archive match those in the JOSS paper.

Do you have a preferred way for both places?

**Comment by nicrie** [comment]
Thanks @slevang !

Regarding `nbstripout`: Currently, only README and quickstart.ipynb should be tested through CI, not the Examples. Since the examples are read in using the .rst (and not the .ipynb) files

https://github.com/xarray-contrib/xeofs/blob/48e63a09579ff559908183ea2f13032886d64290/docs/conf.py#L90-L98

they should be rendered independently of `nbstripout`, as long as they have been run locally once. I might be mistaken, but `quickstart.ipynb` should be executed anew by readthedocs, so we should detect faulty code. Now that I write it, however, I'm unsure, because while RTD would fail, CI wouldn't... :confused:

File: pyproject.toml (hunk 1 above)
Comment (slevang) [reply pyproject.toml]: Ok good point, I reverted back to full name so there are no conflicts.

**Comment by slevang** [comment]
Ok I think I must have run `pre-commit run --all-files` instead of letting the hooks run on commit, and I can confirm that this does try to strip everything in `/docs`. But it looks like if you let it run on commit, it will only touch notebooks you've changed in the workspace. That's a little odd but should be fine.

**Pull request merged**